fdbrpc/ReplicationPolicy.cpp

```diff
@@ -84,7 +84,7 @@ bool IReplicationPolicy::validateFull(
 		auto	missingEntry = totalSolution[lastSolutionIndex];
 		totalSolution[lastSolutionIndex] = totalSolution.back();
 		totalSolution.pop_back();
-		for (int index = 0; index < solutionSet.size(); index ++) {
+		for (int index = 0; index < solutionSet.size() && index < totalSolution.size(); index ++) {
```

Comment:
It almost looks to me like this should just be `index < totalSolution.size()`, omitting the check against `solutionSet.size()`. If I'm understanding correctly, it's removing each entry one at a time and performing some checks on the remaining set. If we stop at `solutionSet.size()`, we would be excluding the removal of servers added from `alsoServers`. That doesn't seem like the intent, though because we start by removing the last server from the set, which comes from `alsoServers`.

Reply:
I agree that it doesn't seem like the intent, but tests in this code fail if we check the removal of `alsoServers`.  Honestly, I don't understand given the tests and the comments what the actual contract of this function is, because they don't seem to match up.  On the plus side, it seems like `validateFull` is only called from tests, so this isn't important to normal fdbserver behavior.

And we still need to check both vectors, as if `alsoServers` is empty, then the `pop_back()` above will remove an element from totalSolution, making totalSolution.size() less than solutionSet.size().

Reply:
I realize I made an incorrect statement. We start by removing the last thing that occurs in solutionSet, not from alsoServers. So it does seem plausible that the intent was to only remove those things that were in solutionSet, in which case what you've written looks good.

Reply:
Although, it does seem like if there are entries in `alsoServers`, then this code will end up removing one of them (when `index == solutionSet.size()-1`). That could be avoided by writing `index < solutionSet.size() - 1` instead of checking against `solutionSet.size()` and `totalSolution.size()`, but I don't know if that's the right thing to do or not. I'm happy to leave this as is you've written it and address that particular question later.